/*
 * This file was automatically generated by EvoSuite
 * Thu Oct 28 09:30:44 UTC 2021
 */

package org.apache.commons.lang3;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.shaded.org.mockito.Mockito.*;
import static org.evosuite.runtime.EvoAssertions.*;
import java.util.Random;
import org.apache.commons.lang3.RandomStringUtils;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.evosuite.runtime.ViolatedAssumptionAnswer;
import org.junit.runner.RunWith;

@RunWith(EvoRunner.class) @EvoRunnerParameters(useVFS = true, useJEE = true) 
public class RandomStringUtils_ESTest extends RandomStringUtils_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test0()  throws Throwable  {
      RandomStringUtils randomStringUtils0 = new RandomStringUtils();
      int int0 = 0;
      boolean boolean0 = true;
      boolean boolean1 = true;
      char[] charArray0 = new char[0];
      Random random0 = mock(Random.class, new ViolatedAssumptionAnswer());
      RandomStringUtils.random(0, 0, 0, true, true, charArray0, random0);
      RandomStringUtils.randomAlphabetic(0);
      Random random1 = mock(Random.class, new ViolatedAssumptionAnswer());
      doReturn(0).when(random1).nextInt(anyInt());
      // Undeclared exception!
      RandomStringUtils.random(1, 1, (-1351), true, true, charArray0, random1);
  }
}
